tests/integration/istio_test.go

```diff
@@ -0,0 +1,496 @@
+// Copyright Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package integration
+
+import (
+	"context"
+	"encoding/json"
+	"strings"
+	"time"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	v1alpha1 "maistra.io/istio-operator/api/v1alpha1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	"istio.io/istio/pkg/ptr"
+)
+
+var _ = Describe("Istio resource", Ordered, func() {
+	const (
+		istioName         = "test-istio"
+		istioNamespace    = "istio-test"
+		workloadNamespace = "istio-test-workloads"
+
+		istioVersion = "latest"
+		oldVersion   = "v1.20.0"
+		newVersion   = "v1.20.1" // TODO: get these versions from versions.yaml
+
+		gracePeriod = 30 * time.Second
+		pilotImage  = "maistra.io/test:latest"
+	)
+	istioKey := client.ObjectKey{Name: istioName}
+	istio := &v1alpha1.Istio{}
+
+	SetDefaultEventuallyTimeout(30 * time.Second)
+	SetDefaultEventuallyPollingInterval(time.Second)
+
+	SetDefaultConsistentlyDuration(30 * time.Second)
+	SetDefaultConsistentlyPollingInterval(time.Second)
+
+	ctx := context.Background()
+
+	namespace := &corev1.Namespace{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: istioNamespace,
+		},
+	}
+
+	BeforeAll(func() {
+		Step("Creating the Namespace to perform the tests")
+		Expect(k8sClient.Create(ctx, namespace)).To(Succeed())
+	})
+
+	AfterAll(func() {
+		// TODO(user): Attention if you improve this code by adding other context test you MUST
+		// be aware of the current delete namespace limitations.
+		// More info: https://book.kubebuilder.io/reference/envtest.html#testing-considerations
```

Comment:
following this link, it states that we should suffix namespaces with `-%d` because namespaces can never be deleted properly and this avoids conflicts. We don't seem to be doing that, so I'm wondering how useful this comment is in the end

Reply:
We currently run the istio and istiorevision tests in different namespaces. We only create/delete each namespace once, so this isn't currently a problem. But let's improve this in another PR.

Reply:
sounds good